### DISCUSSION TITLE
Changing the Onedrive account - Which commands?

### DISCUSSION THREAD

**Discussion by onkel-ede** [abraunegg/onedrive#1752] [Q&A] [answered]
Onedrive works perfectly for me. Now I have read that the Onedrive API cannot handle 2FA. I use Microsoft365-family and still have one free account. I would like to secure my current main account with 2FA because of the account data and subscription settings. I would like to move the Onedrive data to the other account and set it up without 2FA. How can I make the move under Linux?

**Comment by abraunegg** [comment, marked as answer]
@onkel-ede 
To change the account being used and migrate your data, adapt the following process:
* stop any systemd or service process you have running if applicable. If you are using Ubuntu or similar, and installed from a package make sure you have understood the following: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages
* logout from your existing account via: `onedrive --logout`
* migrate your data online (old account -> new account) if this is possible - as this will help to avoid having to re-upload the data
* authorise the application with the new account via: `onedrive`
* test your new account access via: `onedrive --synchronize --dry-run --verbose --local-first <any-other options>`
* sync your local data to ensure a consistent state via: `onedrive --synchronize --verbose --local-first <any-other options>`
* enable any systemd or service if applicable

**Reply by onkel-ede** [reply]
@abraunegg: Thank you for your help. Works great!
